usaspending_api/disaster/v2/views/federal_account/spending.py

```diff
@@ -0,0 +1,230 @@
+                            then=F("gross_outlay_amount_by_program_object_class_cpe"),
+                        ),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+            "total_budgetary_resources": Coalesce(
+                Sum("treasury_account__gtas__budget_authority_appropriation_amount_cpe"), 0
+            ),
+        }
+
+        # Assuming it is more performant to fetch all rows once rather than
+        #  run a count query and fetch only a page's worth of results
+        return (
+            FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters)
+            .values(
+                "treasury_account__federal_account__id",
+                "treasury_account__federal_account__federal_account_code",
+                "treasury_account__federal_account__account_title",
+            )
+            .annotate(**annotations)
+            .values(*annotations.keys())
+        )
+
+    @property
+    def award_queryset(self):
+        filters = [
+            Q(submission__reporting_period_start__gte=self.reporting_period_min),
+            Q(
+                Q(
+                    Q(submission__reporting_period_end__lte=self.recent_monthly_submission["submission_reveal_date"])
+                    & Q(submission__quarter_format_flag=False)
+                )
+                | Q(
+                    Q(submission__reporting_period_end__lte=self.recent_quarterly_submission["submission_reveal_date"])
+                    & Q(submission__quarter_format_flag=True)
+                )
+            ),
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(treasury_account__isnull=False),
+            Q(treasury_account__federal_account__isnull=False),
+        ]
+        annotations = {
+            "fa_code": F("treasury_account__federal_account__federal_account_code"),
+            "count": Count("treasury_account__tas_rendering_label", distinct=True),
+            "description": F("treasury_account__account_title"),
+            "code": F("treasury_account__tas_rendering_label"),
+            "id": F("treasury_account__treasury_account_identifier"),
+            "fa_description": F("treasury_account__federal_account__account_title"),
+            "fa_id": F("treasury_account__federal_account_id"),
+            "obligation": Coalesce(Sum("transaction_obligated_amount"), 0),
+            "outlay": Coalesce(
+                Sum(
+                    Case(
+                        When(
+                            Q(
+                                submission__reporting_fiscal_year=self.recent_monthly_submission[
+                                    "submission_fiscal_year"
+                                ],
+                                submission__reporting_fiscal_period=self.recent_monthly_submission[
+                                    "submission_fiscal_month"
+                                ],
```

Comment:
So this is going to fail until the end of July making it hard to test.  Do we want to make it more defensive or just ensure we have at least one available monthly submission window for June/early-July?